controllers/nutanixmachine_controller.go

```diff
@@ -358,8 +374,18 @@ func (r *NutanixMachineReconciler) reconcileDelete(rctx *nctx.MachineContext) (r
 			} else {
 				log.V(1).Info(fmt.Sprintf("no task UUID found on VM %s. Starting delete.", *vm.Spec.Name))
 			}
+
+			if rctx.NutanixClientV4 != nil {
+				if err := detachVolumeGroupsFromVM(ctx, rctx.NutanixClientV4, vmUUID); err != nil {
+					errorMsg := fmt.Errorf("failed to detach volume groups from VM %s with UUID %s: %v", vmName, vmUUID, err)
+					conditions.MarkFalse(rctx.NutanixMachine, infrav1.VMProvisionedCondition, infrav1.DeletionFailed, capiv1.ConditionSeverityWarning, errorMsg.Error())
```

Comment:
should we add a new condition as volume group detach failed instead of vm Deletion failed since we set same condition on deleteVM failed as well

Reply:
added one